File: the-lambda-trilogy/python/the_lambda_trilogy/the_fat_lambda_stack.py
from aws_cdk import (
    aws_lambda as _lambda,
    aws_apigateway as api_gw,
    core
)


class TheFatLambdaStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        ###
        # Even though all logic lives in the same file, we have 3 separate lambda functions
        ###

        add_lambda = _lambda.Function(self, "addLambdaHandler",
                                      runtime=_lambda.Runtime.PYTHON_3_8,
                                      handler="fatlambda.add",
                                      code=_lambda.Code.from_asset("lambda_fns/the_fat_lambda")
                                      )

        subtract_lambda = _lambda.Function(self, "subtractLambdaHandler",
                                           runtime=_lambda.Runtime.PYTHON_3_8,
                                           handler="fatlambda.subtract",
                                           code=_lambda.Code.from_asset("lambda_fns/the_fat_lambda")
                                           )

        multiply_lambda = _lambda.Function(self, "multiplyLambdaHandler",
                                           runtime=_lambda.Runtime.PYTHON_3_8,
                                           handler="fatlambda.multiply",
                                           code=_lambda.Code.from_asset("lambda_fns/the_fat_lambda")
                                           )

        ###
        # All functions have their own endpoint defined on our gateway
        ##

        api = api_gw.LambdaRestApi(self, 'fatLambdaAPI',
                                   handler=add_lambda,
                                   proxy=False
                                   )

        api.root.resource_for_path('add').add_method('GET', api_gw.LambdaIntegration(add_lambda))
        api.root.resource_for_path('subtract').add_method('GET', api_gw.LambdaIntegration(subtract_lambda))
        api.root.resource_for_path('multiply').add_method('GET', api_gw.LambdaIntegration(multiply_lambda))

File: the-lambda-trilogy/typescript/lib/the-fat-lambda-stack.ts
import * as cdk from '@aws-cdk/core';
import lambda = require('@aws-cdk/aws-lambda');
import apigw = require('@aws-cdk/aws-apigateway');

export class TheFatLambdaStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    /**
     * Lambdas defined individually even though code is all in one file
     */

    const addLambda = new lambda.Function(this, 'addLambdaHandler', {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromAsset('lambda-fns/fat-lambda'),
      handler: 'fat-lambda.add',                
    });

    const subtractLambda = new lambda.Function(this, 'subtractLambdaHandler', {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromAsset('lambda-fns/fat-lambda'),
      handler: 'fat-lambda.subtract',                
    });

    const multiplyLambda = new lambda.Function(this, 'multiplyLambdaHandler', {
      runtime: lambda.Runtime.NODEJS_12_X,
      code: lambda.Code.fromAsset('lambda-fns/fat-lambda'),
      handler: 'fat-lambda.multiply',                
    });


    /**
     * Routes defined individually on API Gateway
     */
    
    let gateway = new apigw.LambdaRestApi(this, 'FatLambdaAPI', {
      handler: addLambda,
      proxy: false
    });

    gateway.root.resourceForPath('add').addMethod('GET', new apigw.LambdaIntegration(addLambda));
    gateway.root.resourceForPath('subtract').addMethod('GET', new apigw.LambdaIntegration(subtractLambda));
    gateway.root.resourceForPath('multiply').addMethod('GET', new apigw.LambdaIntegration(multiplyLambda));
  }
}
